controllers/providers/assistant/gslb.go

```diff
@@ -58,8 +58,7 @@ func NewGslbAssistant(client client.Client, k8gbNamespace string, edgeDNSServers
 	}
 }
 
-// CoreDNSExposedIPs retrieves list of IP's exposed by CoreDNS
-func (r *Gslb) CoreDNSExposedIPs() ([]string, error) {
+func (r *Gslb) GetCoreDNSService() (*corev1.Service, error) {
```

Comment:
I like this separation of responsibilities. One function to fetch the service and another one to fetch the IP address